cdap-distributions/bin/prep_and_stage_bundles.sh

```diff
@@ -0,0 +1,138 @@
+REMOTE_USER=${1}                                        ### remote user
+REMOTE_HOST=${2:-127.0.0.1}                             ### remote host
+REMOTE_INCOMING_DIR=${3}                                ### target directory on remote host
+BUILD_PACKAGE=${BUILD_PACKAGE:-cdap}
+REMOTE_BASE_DIR="incoming"
+USER=${USER:-bamboo}
+TMP_DIR='/tmp/bundles'
+S3STG_DIR=${S3STG_DIR:-/data/s3stg} # change it #####
+DEB_BUNDLE="${BUILD_PACKAGE}-deb-bundle"
+RPM_BUNDLE="${BUILD_PACKAGE}-rpm-bundle"
+DEB_BUNDLE_DIR="${TMP_DIR}/${DEB_BUNDLE}"
+RPM_BUNDLE_DIR="${TMP_DIR}/${RPM_BUNDLE}"
+DISTRIBUTED_DEB_BUNDLE="${BUILD_PACKAGE}-distributed-deb-bundle"
+DISTRIBUTED_RPM_BUNDLE="${BUILD_PACKAGE}-distributed-rpm-bundle"
+DEB_BUNDLE_TGZ="${DISTRIBUTED_DEB_BUNDLE}-${VERSION}.tgz"
+RPM_BUNDLE_TGZ="${DISTRIBUTED_RPM_BUNDLE}-${VERSION}.tgz"
+DEB_BUNDLE_TGZ_PATH="${DEB_BUNDLE_DIR}/${DEB_BUNDLE_TGZ}"
+RPM_BUNDLE_TGZ_PATH="${RPM_BUNDLE_DIR}/${RPM_BUNDLE_TGZ}"
+S3STG_INCOMING="${S3STG_DIR}/incoming"
+S3STG_INCOMING_PROJ="${S3STG_INCOMING}/${BUILD_PACKAGE}"
+STG_DIR="${S3STG_INCOMING}/downloads/co/cask/${BUILD_PACKAGE}"
+STG_DEB_DIR="${STG_DIR}/${DISTRIBUTED_DEB_BUNDLE}/${VERSION}"
+STG_RPM_DIR="${STG_DIR}/${DISTRIBUTED_RPM_BUNDLE}/${VERSION}"
+
+#############################
+# output trimmer
+decho () {
+  if [[ "${DEBUG}" == 'yes' ]]; then
+    echo ${*}
+  else
+    RSYNC_QUIET='--quiet'
+  fi
+}
+
+# Help function
+function HELP {
+  echo -e \\n"Help documentation for ${BOLD}${SCRIPT}.${NORM}"\\n
+  echo -e "${REV}Basic usage:${NORM} ${BOLD}$SCRIPT${NORM} <remote_user> <remote_host> <remote_target_directory"\\n
+  echo -e "${REV}-h${NORM}  --Displays this help message. No further functions are performed."\\n
+  exit 1
+}
+
+die ( ) { echo ; echo "ERROR: ${*}" ; echo ; exit 1; }
+
+######################################################################################
+
+clean() { test -d ${TMP_DIR} && rm -rf ${TMP_DIR}; };
+
+if [ $# -lt 2 ]; then
+  die "You need to pass project and version to ${_script}"$'\n\n'"Example: ./${_script} cdap 3.0.0"
+fi
+
+#########################################################################
+create_bundles() {
+
+   ### PREP
+   # make sure bundle directories exist
+   mkdir -p ${STG_DIR} || die "Unable to create STG_DIR"
+   mkdir -p ${DEB_BUNDLE_DIR} ${RPM_BUNDLE_DIR} || die "Unable to create bundle processing directories"
+   mkdir -p ${STG_DEB_DIR} ${STG_RPM_DIR} || die "Unable to create bundle staging directories"
+   OUTGOING_DIR=${BUILD_PACKAGE}/${VERSION}
+   echo ${OUTGOING_DIR}
+   
+   ### FIND ARTIFACTS
+   cd ${S3STG_INCOMING_PROJ} || die "Unable to cd to ${S3STG_INCOMING_PROJ} directory"
+   find . -name "${BUILD_PACKAGE}-*${VERSION}-*.deb" -exec cp '{}' ${DEB_BUNDLE_DIR}/. \; || die "Unable to copy Debian ${VERSION} packages to TMP directory"
+   find . -name "${BUILD_PACKAGE}-*${VERSION}-*.rpm" -exec cp '{}' ${RPM_BUNDLE_DIR}/. \; || die "Unable to copy ${VERSION} RPMs to TMP directory"
```

Comment:
Why are you using `find .` from a subdirectory rather than the top-level directory? Even better, don't use `find` at all. It's a waste of time and energy when you have a known filename pattern in a known set of directory locations.

From the top of the cdap checkout:
```
cp -f */target/${BUILD_PACKAGE}-*${VERSION}-*.rpm ${RPM_BUNDLE_DIR}
```

Or even simply:
```
cp -f */target/*.rpm ${RPM_BUNDLE_DIR}
```